### ISSUE TITLE
xml2json:  missing root node

### ISSUE THREAD

**Issue by rickhg12hs** [simonrupf/convert2json#48]
I'm no XML/JSON expert, but I'm wondering why the XML root node is missing in the JSON.

test.xml
```lang-xml
<?xml version="1.0" encoding="UTF-8"?>
<root>
    
    <node1/>
    
    <node2/>
</root>
```
```lang-bash
$ xml2json test.xml 
{"node1":{},"node2":{}}
```
Should the output be something like:
```lang-json
{"root": {"node1": {}, "node2": {}}}
```
?

Or even possibly one of:
```lang-json
{"root": [{"node1": {}}, {"node2": {}}]}
{"root": {"node1": null, "node2": null}}
{"root": [{"node1": null}, {"node2": null}]}
```
?

**Comment by simonrupf** [comment]
Indeed, serde_xml_rs, the library used in the XML deserialization, doesn't preserve it's name. There is a comment regarding this in the [caveats section of their docs](https://docs.rs/serde-xml-rs/latest/serde_xml_rs/#caveats):

> Other notable differences:
> 
> -  XML requires a named root node.

When used to deserialize into a single known type, that isn't much of an issue, but if one had different types and wanted to return the correct one based on the root-elements name it might not work as expected. This issue is likely related:
https://github.com/RReverser/serde-xml-rs/issues/139

With the current version of the library, I don't see a way to change this behavior, short of doing additional parsing in this projects code. At least any attributes of the root node will get preserved:

```shell
$ cat test.xml 
<?xml version="1.0" encoding="UTF-8"?>
<root root-attr="something">
    <node1/>
    <node2/>
</root>
$ xml2json test.xml
{"node1":{},"node2":{},"root-attr":"something"}
```

I could start a "caveats" section in the README.md, documenting problematic corner cases like these, for the time being.

**Comment by simonrupf** [comment]
This issue is now addressed in release 1.0.0, along with switching to the maintained quick-xml library.

This is a major, breaking change and the resulting JSON will have a different
structure. In particular:
  - document root node gets preserved as the outermost objects only key (this issue):
    `<root/>` becomes `{"root": {}}` (instead of `{}`)
  - attributes start with an @-character to distinguish them from inner tags:
    `<tag attribute="value"><inner/></tag>` becomes
    `{"tag":{"@attribute":"value","inner":{}}}`
    (instead of `{"attribute":"value","inner":{}}`)
  - text nodes are now called `$text` instead of `$value`:
    `<key>value</key>` becomes `{"key":{"$text":"value"}}`
    (instead of `{"$value":"value"}`)
